web/frontend/src/mocks/handlers.ts

```diff
@@ -153,53 +176,105 @@ export const handlers = [
       ...mockElections.get(ElectionID as string),
       Status,
     });
-    await new Promise((r) => setTimeout(r, 1000));
 
     return res(ctx.status(200), ctx.text('Action successfully done'));
   }),
 
   rest.put(endpoints.editShuffle(':ElectionID'), async (req, res, ctx) => {
     const { ElectionID } = req.params;
+    const id = sessionStorage.getItem('id');
+    const userRole = mockUserDB.find(({ sciper }) => sciper === id).role;
+
+    await new Promise((r) => setTimeout(r, 1000));
+
+    if (userRole !== ROLE.Admin && userRole !== ROLE.Operator) {
+      return res(
+        ctx.status(403),
+        ctx.json({ message: 'You are not authorized to create an election' })
+      );
+    }
+
     mockElections.set(ElectionID as string, {
       ...mockElections.get(ElectionID as string),
       Status: STATUS.ShuffledBallots,
     });
-    await new Promise((r) => setTimeout(r, 1000));
 
     return res(ctx.status(200), ctx.text('Action successfully done'));
   }),
 
   rest.put(endpoints.editDKGActors(':ElectionID'), async (req, res, ctx) => {
     const { ElectionID } = req.params;
+    const id = sessionStorage.getItem('id');
+    const userRole = mockUserDB.find(({ sciper }) => sciper === id).role;
+
+    await new Promise((r) => setTimeout(r, 1000));
+
+    if (userRole !== ROLE.Admin && userRole !== ROLE.Operator) {
+      return res(
+        ctx.status(403),
+        ctx.json({ message: 'You are not authorized to create an election' })
+      );
+    }
+
     mockElections.set(ElectionID as string, {
       ...mockElections.get(ElectionID as string),
       Result: mockResults.get(ElectionID as string),
       Status: STATUS.ResultAvailable,
     });
-    await new Promise((r) => setTimeout(r, 1000));
 
     return res(ctx.status(200), ctx.text('Action successfully done'));
   }),
 
   rest.get(endpoints.ENDPOINT_USER_RIGHTS, async (req, res, ctx) => {
+    const id = sessionStorage.getItem('id');
+    const userRole = mockUserDB.find(({ sciper }) => sciper === id).role;
+
     await new Promise((r) => setTimeout(r, 1000));
 
+    if (userRole !== ROLE.Admin) {
```

Comment:
you should refactor this code with a generic `checkAuth(roles): boolean` function that you can call in your handlers.